src/mlpack/methods/reinforcement_learning/replay/prioritized_replay.hpp

```diff
@@ -95,13 +117,33 @@ class PrioritizedReplay
    * @param reward Given reward.
    * @param nextState Given next state.
    * @param isEnd Whether next state is terminal state.
+   * @param discount The discount parameter.
    */
-  void Store(const StateType& state,
+  void Store(StateType state,
              ActionType action,
              double reward,
-             const StateType& nextState,
-             bool isEnd)
+             StateType nextState,
+             bool isEnd,
+             const double& discount)
   {
+    nStepBuffer.push_back({state, action, reward, nextState, isEnd});
+
+    // Single step transition is not ready.
+    if (nStepBuffer.size() < nSteps)
+      return;
+
+    // To keep the queue size fixed to nSteps.
+    if (nStepBuffer.size() > nSteps)
+      nStepBuffer.erase(nStepBuffer.begin());
```

Comment:
I think since this is a deque, we could use `pop_front()`

Reply:
Yeah right. Thanks for this